### ISSUE TITLE
Samsung Galaxy Tab S6 Lite (SM-P610) Recovery flash never finishes

### ISSUE THREAD

**Issue by lobneroO** [Benjamin-Dobell/Heimdall#506]
I tried to use Heimdall to flash a recovery onto my Samsung Galaxy Tab S6 Lite (SM-P610). My PC is running Arch Linux, I have Heimdall v1.4.2. 
I followed the instructions given by LineageOS: https://wiki.lineageos.org/devices/gta4xlwifi/install

The actual flashing part is describe in "Installing a custom recovery using heimdall".
I came to execute step 3 and used `heimdall flash --RECOVERY <recovery_filename>.img --no-reboot`

This partly worked: the flashing process on my Arch machine fully finished, however on my tablet, the progress bar never went to full; it stops short just shy of full. It may actually even be full, but I don't think so - anyway, in the lineage instructions it says that a status message would indicate it finishing somewhere at the top and that certainly never happened.

However I went ahead anyway, unplugged the usb cable and used the Power and Volume Down key to restart (in the tablet screen this said it would cancel the flashing) and then on reboot I immediately stopped the boot process by holding Power and Volume Down keys again until the device turned off.
Afterwards I could start the recovery mode by connecting the USB cable again and then hold Power and Volume Up keys. The Lineage installation after that worked as described in their manual.

I wanted to post this somewhere, since it might help others. Not sure if this is the best place to do so, but I didn't know any better place.

tl;dr: Heimdall's flashing process on the tablet never notices that it actually finished. Once it's finished, disconnect the usb cable, hold Power + Volume Down to restart, then again hold Power + Volume Down to cancel the boot process. Finally, reconnect the usb cable and hold Power + Volume Up to start the recovery mode.

**Comment by Grimler91** [comment]
> however on my tablet, the progress bar never went to full

This is intended behaviour with the `--no-reboot` flag, it doesn't go to full since you are indicating that you are planning to run additional commands, or flash more files.

I guess lineage suggests `--no-reboot` to make it possible for users to reboot into download mode when they are ready (rather than it auto-rebooting).

> says that a status message would indicate it finishing somewhere at the top and that certainly never happened

I haven't seen that type of message either (errors on failed transfers appear there in the corner though)
